dot/network/host.go

```diff
@@ -187,7 +187,7 @@ func newHost(ctx context.Context, cfg *Config) (*host, error) {
 	}
 
 	// create libp2p host instance
-	h, err := libp2p.New(ctx, opts...)
+	h, err := libp2p.New(opts...)
```

Comment:
Since we don't pass that context anymore, how is context stuff handled? :thinking:
I don't see it passed elsewhere :thinking:

Reply:
I went quite a bit in the rabbit hole, it looks like the context is now created in

https://github.com/libp2p/go-libp2p/blob/5c297a87002a04a68143c59167502a9be6c73f2f/p2p/net/swarm/swarm.go#L148

and it's canceled by some other function calls. Maybe you can have a quick look where this swarm cancel function is called? :thinking: So we can shutdown libp2p gracefully when stopping our network service

Reply:
```
Breaking Changes
The context.Context in the host constructor (libp2p.New) was removed (https://github.com/libp2p/go-libp2p/pull/1190). Use Host.Close() to shut down a host.
```
Explicit closure of the host rather than using contexts! Very nice!